source/pod.go

```diff
@@ -71,6 +72,36 @@ func NewPodSource(
 			},
 		},
 	)
+
+	// Transformer is used to reduce the memory usage of the informer.
+	// The pod informer will otherwise store a full in-memory, go-typed copy of all pod schemas in the cluster.
+	// If watchList is not used it will not prevent memory bursts on the initial informer sync.
+	podInformer.Informer().SetTransform(func(i interface{}) (interface{}, error) {
```

Comment:
I wonder how this is going to work with FQDN template, we could do some pre-processing and attach it as a field ?